http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/ProxyService.scala

```diff
@@ -124,6 +125,8 @@ class ProxyService(
           val res = for {
             now <- nowInstant
             userInfo <- googleOauth2Service.getUserInfoFromToken(key)
+            userOpt <- samDAO.getUserSubjectIdFromToken(key)
+            _ <- IO.fromOption(userOpt)(AuthenticationError(Some(userInfo.userEmail)))
```

Comment:
Trying to remember what happens if an exception is thrown inside the `CacheLoader`. I think it gets re-thrown as `ExecutionException` in the `cache.get()` call?